src/modules/kvs/kvs.c

```diff
@@ -3026,6 +3070,15 @@ int mod_main (flux_t *h, int argc, char **argv)
         flux_log_error (h, "flux_reactor_run");
         goto done;
     }
+    if (zhashx_size (ctx->requests) > 0) {
+        /* anything that has not yet completed gets an ENOSYS */
+        const flux_msg_t *msg = zhashx_first (ctx->requests);
+        while (msg) {
+            if (flux_respond_error (ctx->h, msg, ENOSYS, NULL) < 0)
+                flux_log_error (ctx->h, "%s: flux_respond_error", __FUNCTION__);
+            msg = zhashx_next (ctx->requests);
+        }
+    }
```

Comment:
Suggestion: log something at LOG_ERR for each request like:`failing pending <topic> request`.

Also, maybe in all the KVS tests, ensure that this message is not logged just to be sure we don't have a `request_tracking_add()` without a corresponding `request_tracking_remove()` somewhere, since the result of that could be quite bad - like terminating an unrelated RPC with ENOSYS and that will be hard to track down.

Could also add the size of this list to the `flux module stats` output since that may be useful in debugging.

Reply:
good call, caught a corner case and possibly another bug unrelated to this PR